File: LayoutTests/http/tests/resourceLoadStatistics/resources/set-all-kinds-of-cookies.php
<?php
header("HTTP/1.1 200 OK");

setcookie("persistentCookie", "1", time() + 300, "/");
setcookie("persistentSecureCookie", "1", time() + 300, "/", "", TRUE);
setcookie("persistentHttpOnlyCookie", "1", time() + 300, "/", "", FALSE, TRUE);
setcookie("persistentSameSiteLaxCookie", "1", ["expires" => time() + 300, "path" => "/", "samesite" => "Lax"]);
setcookie("persistentSameSiteStrictCookie", "1", ["expires" => time() + 300, "path" => "/", "samesite" => "Strict"]);

setcookie("sessionCookie", "1", "", "/");
setcookie("sessionSecureCookie", "1", "", "/", "", TRUE);
setcookie("sessionHttpOnlyCookie", "1", "", "/", "", FALSE, TRUE);
setcookie("sessionSameSiteLaxCookie", "1", ["path" => "/", "samesite" => "Lax"]);
setcookie("sessionSameSiteStrictCookie", "1", ["path" => "/", "samesite" => "Strict"]);
?>

File: LayoutTests/http/tests/resourceLoadStatistics/resources/set-all-kinds-of-cookies.py
#!/usr/bin/env python3

import sys
from datetime import datetime, timedelta

expires = datetime.utcnow() + timedelta(seconds=300)

sys.stdout.write(
    'status: 200\r\n'
    'Set-Cookie: persistentCookie=1; expires={expires} GMT; Max-Age=300; path=/\r\n'
    'Set-Cookie: persistentSecureCookie=1; expires={expires} GMT; Max-Age=300; path=/; secure\r\n'
    'Set-Cookie: persistentHttpOnlyCookie=1; expires={expires} GMT; Max-Age=300; path=/; HttpOnly\r\n'
    'Set-Cookie: persistentSameSiteLaxCookie=1; expires={expires} GMT; Max-Age=300; path=/; SameSite=Lax\r\n'
    'Set-Cookie: persistentSameSiteStrictCookie=1; expires={expires} GMT; Max-Age=300; path=/; SameSite=Strict\r\n'
    'Set-Cookie: sessionCookie=1; path=/\r\n'
    'Set-Cookie: sessionSecureCookie=1; path=/; secure\r\n'
    'Set-Cookie: sessionHttpOnlyCookie=1; path=/; HttpOnly\r\n'
    'Set-Cookie: sessionSameSiteLaxCookie=1; path=/; SameSite=Lax\r\n'
    'Set-Cookie: sessionSameSiteStrictCookie=1; path=/; SameSite=Strict\r\n'
    'Content-Type: text/html\r\n\r\n'.format(expires=expires.strftime('%a, %d-%b-%Y %H:%M:%S'))
)